src/core/lwt_seq.ml

```diff
@@ -0,0 +1,115 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                 Simon Cruanes                                          *)
+(*                                                                        *)
+(*   Copyright 2017 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+
+
+open Lwt.Syntax
+
+type +'a node = Nil | Cons of 'a * 'a t
+
+and 'a t = unit -> 'a node Lwt.t
+
+let empty : 'a t = fun () -> Lwt.return Nil
+
+let return x : 'a t = fun () -> Lwt.return (Cons (x, empty))
+
+let cons x t () = Lwt.return (Cons (x, t))
+
+let rec append seq1 seq2 () =
+  let* seq1 = seq1 () in
+  match seq1 with
```

Comment:
(Not that it should impact this PR, but is there a plan to have `match*` in OCaml soon?)

Reply:
Relevant PR: https://github.com/ocaml/ocaml/pull/1955

Reply:
I'm ok with using the `let`-operators. This is mostly a matter of style. Are there any strong opinions on the matter?

I guess the main reason to use `>>=` instead of `let*` would be to keep compatibility with old ocamls, but we're already `>= 4.8` which is when the binds where introduced.

Reply:
My suggestion for removing the `let*` uses in these places was because it avoids introducing a name that is immediately matched on, but I don't have a strong opinion either way :-)